spec/elastic_apm/util/deep_dup_spec.rb

```diff
@@ -0,0 +1,49 @@
+
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+module ElasticAPM
+  RSpec.describe Util::DeepDup do
+    context 'hashes' do
+      it 'makes a copy' do
+        obj1 = Object.new
+        obj2 = Object.new
+        orig = { a: obj1, nested: { b: obj2 } }
+
+        expect(orig.dup[:a]).to be obj1
+        expect(orig.dup[:nested][:b]).to be obj2
```

Comment:
what's the point of testing these two lines?

Reply:
I wanted to be sure but you're right. No point in testing Ruby itself.